### PR TITLE
Add browser to examples

### PR DESCRIPTION
**Pull request by pusztaienike**

### CODE REVIEW

File: examples/sn-react-browser/src/hooks/use-current-user.ts [hunk 1]
```diff
@@ -0,0 +1,17 @@
+import { useEffect, useState } from 'react'
+import { User } from '@sensenet/default-content-types'
+import { useRepository } from '@sensenet/hooks-react'
+
+/**
+ * Custom hook that will return with the current user.
+ */
+export const useCurrentUser = () => {
+  const repo = useRepository()
```
Comment (B3zo0) [inline examples/sn-react-browser/src/hooks/use-current-user.ts]: Remove this file.

File: examples/sn-react-browser/src/hooks/use-injector.ts [hunk 2]
```diff
@@ -0,0 +1,4 @@
+import { useContext } from 'react'
+import { InjectorContext } from '../context/injector-context-provider'
+
+export const useInjector = () => useContext(InjectorContext)
```
Comment (B3zo0) [inline examples/sn-react-browser/src/hooks/use-injector.ts]: Remove this file.

File: examples/sn-react-browser/src/hooks/use-selection-service.ts [hunk 3]
```diff
@@ -0,0 +1,8 @@
+import { useContext, useState } from 'react'
+import { InjectorContext } from '../context/injector-context-provider'
+import { SelectionService } from '../service/selection-service'
+
+export const useSelectionService = () => {
+  const [selectionService] = useState(useContext(InjectorContext).getInstance(SelectionService))
```
Comment (B3zo0) [inline examples/sn-react-browser/src/hooks/use-selection-service.ts]: Remove this file.

File: examples/sn-react-browser/src/index.tsx [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+
+/** Initialize the reducers */
+const appReducer = combineReducers({
+  sensenet: Reducers.sensenet,
+  sensenetDocumentViewer: sensenetDocumentViewerReducer,
+  comments: commentsStateReducer,
+  wopi,
+})
+export type rootStateType = ReturnType<typeof appReducer>
+const diMiddleware = new ReduxDiMiddleware(snInjector)
```
Comment (B3zo0) [inline examples/sn-react-browser/src/index.tsx]: Remove redux logic from here.

File: examples/sn-react-browser/src/service/docview-setting.ts [hunk 5]
```diff
@@ -0,0 +1,186 @@
...
+    return {
+      ...comment,
+      createdBy: { ...comment.createdBy, avatarUrl: `${documentData.hostName}${comment.createdBy.avatarUrl}` },
+    }
+  }
+}
+
+export const getViewerSettings = (repo: Repository) =>
+  new DocumentViewerSettings({
+    regeneratePreviews: async documentData => {
```
Comment (B3zo0) [inline examples/sn-react-browser/src/service/docview-setting.ts]: This file is not needed.

File: examples/sn-react-browser/src/service/repository-manager.ts [hunk 6]
```diff
@@ -0,0 +1,72 @@
+import { Injectable, Injector } from '@furystack/inject'
+import { FormsAuthenticationService, LoginState, Repository } from '@sensenet/client-core'
+import { RepositoryConfiguration } from '@sensenet/client-core/dist/Repository/RepositoryConfiguration'
+import { EventHub } from '@sensenet/repository-events'
+import { RequestCounterService } from './request-counter-service'
+
+@Injectable({ lifetime: 'singleton' })
+export class RepositoryManager {
+  private repos: Map<string, Repository> = new Map()
```
Comment (B3zo0) [inline examples/sn-react-browser/src/service/repository-manager.ts]: Remove this file.

File: examples/sn-react-browser/src/service/request-counter-service.ts [hunk 7]
```diff
@@ -0,0 +1,91 @@
...
+      all: number
+      GET: number
+      POST: number
+    }
+  }
+}
+
+@Injectable({ lifetime: 'singleton' })
+export class RequestCounterService implements IDisposable {
+  private readonly writeInterval: NodeJS.Timeout
```
Comment (B3zo0) [inline examples/sn-react-browser/src/service/request-counter-service.ts]: Remove this file.

File: examples/sn-react-browser/src/service/selection-service.ts [hunk 8]
```diff
@@ -0,0 +1,17 @@
+import { Injectable } from '@furystack/inject'
+import { Disposable, ObservableValue } from '@sensenet/client-utils'
+import { GenericContent } from '@sensenet/default-content-types'
+
+/**
+ * A context service to get/set the active item and the selection
+ */
+@Injectable({ lifetime: 'singleton' })
+export class SelectionService implements Disposable {
+  public activeContent = new ObservableValue<GenericContent | undefined>(undefined)
```
Comment (B3zo0) [inline examples/sn-react-browser/src/service/selection-service.ts]: Remove this file.

File: examples/sn-react-browser/src/store/wopi-actions.ts [hunk 9]
```diff
@@ -0,0 +1,7 @@
+import { Repository } from '@sensenet/client-core'
+
+export const getWopiData = (idOrPath: string | number) => ({
+  type: 'GET_WOPIDATA',
```
Comment (B3zo0) [inline examples/sn-react-browser/src/store/wopi-actions.ts]: Remove this file.

File: examples/sn-react-browser/src/store/wopi-reducer.ts [hunk 10]
```diff
@@ -0,0 +1,18 @@
...
+
+const wopiDataInitialState = {
+  accesstoken: '',
+  actionUrl: '',
+  expiration: 0,
+  faviconUrl: '',
+}
+
+export const wopi: Reducer<ODataWopiResponse> = (state = wopiDataInitialState, action: AnyAction) => {
+  switch (action.type) {
```
Comment (B3zo0) [inline examples/sn-react-browser/src/store/wopi-reducer.ts]: Remove this file.

File: examples/sn-react-browser/src/utils/browser-history.ts [hunk 11]
```diff
@@ -0,0 +1,3 @@
+import { createBrowserHistory } from 'history'
+
+export default createBrowserHistory()
```
Comment (B3zo0) [inline examples/sn-react-browser/src/utils/browser-history.ts]: I'm not sure if we need this. Probably not.

File: examples/sn-react-browser/src/utils/injector-extension.ts [hunk 12]
```diff
@@ -0,0 +1,26 @@
+import { Injector } from '@furystack/inject/dist/Injector'
+import { Repository } from '@sensenet/client-core'
+import { RepositoryConfiguration } from '@sensenet/client-core/dist/Repository/RepositoryConfiguration'
+import { EventHub } from '@sensenet/repository-events'
+import { RepositoryManager } from '../service/repository-manager'
+
+declare module '@furystack/inject/dist/Injector' {
+  /**
```
Comment (B3zo0) [inline examples/sn-react-browser/src/utils/injector-extension.ts]: Remove this file.

File: examples/sn-react-browser/test/mainpanel.test.tsx [hunk 13]
```diff
@@ -0,0 +1,184 @@
...
+  it('should renders correctly', async () => {
+    await act(async () => {
+      wrapper = mount(
+        <RepositoryContext.Provider value={repo}>
+          <MainPanel {...MainPanelProps} />
+        </RepositoryContext.Provider>,
+      )
+    })
+
+    expect(wrapper.update()).toMatchSnapshot()
```
Comment (B3zo0) [inline examples/sn-react-browser/test/mainpanel.test.tsx]: This test is useless. Please remove it with the snapshot as well.

File: examples/sn-react-browser/test/repository-provider.test.tsx [hunk 14]
```diff
@@ -0,0 +1,15 @@
+import { mount } from 'enzyme'
+import React from 'react'
+import { act } from 'react-dom/test-utils'
+import { RepositoryProvider } from '../src/context/repository-provider'
+
+describe('Repository Provider', () => {
+  it('Repository Provider snapshot', async () => {
+    let wrapper: any
```
Comment (B3zo0) [inline examples/sn-react-browser/test/repository-provider.test.tsx]: Remove this file with the snapshot as well.

File: yarn.lock [hunk 15]
```diff
@@ -1224,11 +1224,31 @@
     "@furystack/utils" "^1.0.1"
     reflect-metadata "^0.1.13"
 
+"@furystack/inject@^4.0.2", "@furystack/inject@^4.0.4":
```
Comment (B3zo0) [inline yarn.lock]: I  suspect no changes here after the package.json cleanup.

File: examples/sn-react-browser/package.json [hunk 16]
```diff
@@ -0,0 +1,60 @@
...
+    "@types/react-dom": "^16.9.4",
+    "@types/react-router": "^5.1.2",
+    "@types/react-router-dom": "^5.1.2",
+    "react": "^16.12.0",
+    "react-dom": "^16.12.0",
+    "react-scripts": "^3.2.0",
+    "typescript": "^3.7.2"
+  },
+  "dependencies": {
+    "@material-ui/core": "^4.7.0",
```
Comment (B3zo0) [inline examples/sn-react-browser/package.json]:
```suggestion
    "@material-ui/core": "^4.6.1",
```
please change this back to 4.6.1 we will upgrade material ui after the release. Also don't forget to update the yarn.lock

File: examples/sn-react-browser/public/index.html [hunk 17]
```diff
@@ -0,0 +1,15 @@
+<!DOCTYPE html>
+<html lang="en">
+  <head>
+    <meta charset="utf-8" />
+    <link rel="icon" href="%PUBLIC_URL%/favicon.ico" />
+    <meta name="viewport" content="width=device-width, initial-scale=1" />
+    <meta name="theme-color" content="#000000" />
+    <meta name="description" content="Sensenet browser example app" />
+    <title>Sensenet browser example app</title>
```
Comment (herflis) [inline examples/sn-react-browser/public/index.html]:
```suggestion
    <title>sensenet browser example app</title>
```

File: examples/sn-react-browser/src/components/login-form.tsx [hunk 18]
```diff
@@ -0,0 +1,84 @@
...
+  onLogin: (username: string, passowrd: string, repository: string) => void
+
+  /**
+   * An optional error message
+   */
+  error?: string
+}
+
+/**
+ * Login component for Sensenet repositories
```
Comment (herflis) [inline examples/sn-react-browser/src/components/login-form.tsx]:
```suggestion
 * Login component for sensenet repositories
```

File: examples/sn-react-browser/test/app.test.tsx [hunk 19]
```diff
@@ -0,0 +1,9 @@
+import { shallow } from 'enzyme'
+import React from 'react'
+import { App } from '../src/app'
+
+describe('The app layout instance', () => {
+  it('should renders correctly', () => {
```
Comment (herflis) [inline examples/sn-react-browser/test/app.test.tsx]:
```suggestion
  it('should be rendered correctly', () => {
```

File: examples/sn-react-browser/test/full-screen-loader.test.tsx [hunk 20]
```diff
@@ -0,0 +1,9 @@
+import { shallow } from 'enzyme'
+import React from 'react'
+import { FullScreenLoader } from '../src/components/full-screen-loader'
+
+describe('The full screen loader instance', () => {
+  it('should renders correctly', () => {
```
Comment (herflis) [inline examples/sn-react-browser/test/full-screen-loader.test.tsx]:
```suggestion
  it('should be rendered correctly', () => {
```

File: examples/sn-react-browser/test/login-form.test.tsx [hunk 21]
```diff
@@ -0,0 +1,45 @@
...
+import TextField from '@material-ui/core/TextField'
+import { LoginForm } from '../src/components/login-form'
+
+describe('The login instance', () => {
+  const testprop = {
+    onLogin: jest.fn(),
+    error: 'test',
+  }
+
+  it('should renders correctly', () => {
```
Comment (herflis) [inline examples/sn-react-browser/test/login-form.test.tsx]:
```suggestion
  it('should be rendered correctly', () => {
```

File: examples/sn-react-browser/test/login-form.test.tsx [hunk 22]
```diff
@@ -0,0 +1,45 @@
...
+    expect(shallow(<LoginForm {...testprop} />)).toMatchSnapshot()
+  })
+
+  it('should change username state when user change it', async () => {
```
Comment (herflis) [inline examples/sn-react-browser/test/login-form.test.tsx]:
```suggestion
  it('should change username state when user changes it', async () => {
```

File: examples/sn-react-browser/test/login-form.test.tsx [hunk 23]
```diff
@@ -0,0 +1,45 @@
...
+    expect(
+      wrapper
+        .update()
+        .find(TextField)
+        .first()
+        .props().value,
+    ).toEqual('testusername')
+  })
+
+  it('should change repository url state when user change it', async () => {
```
Comment (herflis) [inline examples/sn-react-browser/test/login-form.test.tsx]:
```suggestion
  it('should change repository url state when user changes it', async () => {
```

File: examples/sn-react-browser/test/mainpanel.test.tsx [hunk 24]
```diff
@@ -0,0 +1,188 @@
...
+
+    const downloadicon = wrapper.update().find(CloudDownload)
+    act(() => {
+      ;(downloadicon.prop('onClick') as any)()
+    })
+
+    expect(downloadFile).toBeCalled()
+  })
+
+  it('should open edit mode', async () => {
```
Comment (herflis) [inline examples/sn-react-browser/test/mainpanel.test.tsx]:
```suggestion
  it('should open content edit mode', async () => {
```

File: examples/sn-react-browser/test/navbar.test.tsx [hunk 25]
```diff
@@ -0,0 +1,26 @@
...
+import Button from '@material-ui/core/Button'
+import { RepositoryContext } from '@sensenet/hooks-react'
+import { NavBarComponent } from '../src/components/navbar'
+
+describe('The navbar instance', () => {
+  it('should renders correctly', () => {
+    expect(shallow(<NavBarComponent />)).toMatchSnapshot()
+  })
+
+  it('should logout correctly', () => {
```
Comment (herflis) [inline examples/sn-react-browser/test/navbar.test.tsx]:
```suggestion
  it('should logout the user correctly', () => {
```

**Pull request merged**